libs/astradb/langchain_astradb/vectorstores.py

```
@@ -827,6 +827,62 @@ async def adelete(
         )
         return True
 
+    def delete_by_metadata_filter(
+        self,
+        filter: dict[str, Any],  # noqa: A002
+    ) -> int | None:
+        """Delete all documents matching a certain metadata filtering condition.
+
+        This operation does not use the vector embeddings in any way, it simply
+        removes all documents whose metadata match the provided condition.
+        Use with caution: passing an empty filter dictionary results in
+        completely emptying the vector store.
+
+        Args:
+            filter: Filter on the metadata to apply.
+
+        Returns:
+            An number expressing the amount of deleted documents.
+            This will be None if a `{}` metadata filter condition is passed,
+            implying emptying the store entirely.
+        """
+        self.astra_env.ensure_db_setup()
+        metadata_parameter = self.filter_to_query(filter)
+        del_result = self.astra_env.collection.delete_many(
+            filter=metadata_parameter,
+        )
+        if del_result.deleted_count is not None and del_result.deleted_count >= 0:
+            return del_result.deleted_count
+        return None
```

could we throw an error if `len(filter) == 0` instead of allowing it to delete everything?

---

Fine for me (the reason for allowing `{}` is that we were asked to remove a similar guardrail in astrapy's delete_many. But I appreciate that here the scope is much more limited and safer is presumably better).

I'll change it shortly.